raft/src/main/java/org/apache/kafka/raft/internals/ByteBufferSerde.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.raft.internals;
+
+import java.nio.ByteBuffer;
+import org.apache.kafka.common.protocol.ObjectSerializationCache;
+import org.apache.kafka.common.protocol.Readable;
+import org.apache.kafka.common.protocol.Writable;
+import org.apache.kafka.server.common.serialization.RecordSerde;
+
+public final class ByteBufferSerde implements RecordSerde<ByteBuffer> {
```

Comment:
Seems like a weird hack... we don't want to look at byte buffers, we want to look at control records, right?

Reply:
I agree. I was trying to avoid adding a type parameter to `KafkaMetadataLog` but I found a better way of doing that. I removed `ByteBufferSerde` from the PR.

Reply:
ok